### ISSUE TITLE
 Add "warnings" from prometheus upstream

### ISSUE THREAD

**Issue by jacksontj** [jacksontj/promxy#169]
Split from #141 (replacement of #168)
TLDR; upstream prometheus added a "storage.Warnings" which it uses for non-primary datasource reads. This pr is for adding support for such a thing

TODO:
- [x] support "merging" warnings
- [x] add "make_warnings" option, similar to "ignore_error" except instead of ignoring we make them warnings
- [x] add "make_errors" option, similar to "make_warnings" except we take warnings and make them errors
- [x] add warnings to NodeReplacer

References/links:
- ~~Initial add of warnings to client (https://github.com/prometheus/client_golang/pull/562 )~~
- ~~rework of warnings in api client (https://github.com/prometheus/client_golang/pull/599)~~
- ~~add warnings to series (https://github.com/prometheus/client_golang/pull/603)~~
- ~~labelsName/labelValues warnings (https://github.com/prometheus/prometheus/issues/5661) (https://github.com/prometheus/client_golang/pull/609)~~
- better configuration for warnings (https://github.com/prometheus/prometheus/issues/5662)
